### PR TITLE
docs: Make 100 column limit for code explicit

### PR DESCRIPTION
**Pull request by keith-zephyr**
Zephyr's coding style is based on the Linux kernel coding style, which indicates a preferred limit of 80 columns. Zephyr's .clang-format sets the ColumnLimit to 100. Update the documentation to match the .clang-format parameters.

Signed-off-by: Keith Short <keithshort@google.com>

### CODE REVIEW

File: doc/contribute/guidelines.rst
Comment (evgeniy-paltsev) [inline doc/contribute/guidelines.rst:426]: Just wondering if there is some reasons to limit documentation lines to 80 if we already allow 100 columns for code?

File: doc/contribute/guidelines.rst
Comment (henrikbrixandersen) [inline doc/contribute/guidelines.rst:437]: This is contrary to how we have enforced this in the past. The documented code style takes precedence over clang-format.

**Review by henrikbrixandersen — CHANGES_REQUESTED** [review]

File: MAINTAINERS.yml
Comment (stephanosio) [reply MAINTAINERS.yml:493]: I do not think `.clang-format` belongs under the "Documentation" area.

File: doc/contribute/guidelines.rst
Comment (stephanosio) [reply doc/contribute/guidelines.rst:426]: I am also wondering what is the rationale for setting the line length limit of 80 for docs only.

File: doc/contribute/guidelines.rst
Comment (stephanosio) [reply doc/contribute/guidelines.rst:437]: Indeed, clang-format is a **tool** we use to enforce our coding guidelines. A tool should not take precedence over the very thing that it attempts to facilitate.

**Review by stephanosio — CHANGES_REQUESTED** [review]

File: doc/contribute/guidelines.rst
Comment (keith-zephyr) [reply doc/contribute/guidelines.rst:426]: 80 columns is the currently documented requirement in the [Documentation Guidelines](https://docs.zephyrproject.org/latest/contribute/documentation/guidelines.html#tabs-spaces-and-indenting)

 > Keep the line length for documentation less than 80 characters to make it easier for reviewing in GitHub. Long lines because of URL references are an allowed exception.

File: MAINTAINERS.yml
Comment (keith-zephyr) [reply MAINTAINERS.yml:493]: My thinking is that if we make `.clang-format` the source of truth, then it we want the same maintainers in the loop for `.clang-format` changes as this could require a documentation update to match.

File: doc/contribute/guidelines.rst
Comment (keith-zephyr) [reply doc/contribute/guidelines.rst:437]: When the documented code styles takes precedence, then this potentially means code authors have to hand edit files after running clang-format to conform to the coding style.  This also relies on the reviewers to catch the coding style problems.

I would prefer that we just had a policy to always run clang-format, but that is likely controversial.

File: doc/contribute/guidelines.rst
Comment (stephanosio) [reply doc/contribute/guidelines.rst:426]: I believe that was left at 80 due to an oversight. The docs should have the same line length limit of 100 as code.

**Comment by gmarull** [comment]
Removed my review request, I no longer have interest in clang-format topics, IMO its adoption as a "standard formatting tool" seems difficult because of our decision-making process.

File: doc/contribute/guidelines.rst
Comment (keith-zephyr) [reply doc/contribute/guidelines.rst]: Updated the Doc guidelines.

File: MAINTAINERS.yml
Comment (keith-zephyr) [inline MAINTAINERS.yml:363]: Note - I'm not trying to take over coding guidelines, but I didn't want to add more burden to existing maintainers.

**Review by evgeniy-paltsev — DISMISSED** [review]

File: doc/contribute/guidelines.rst
Comment (keith-zephyr) [reply doc/contribute/guidelines.rst]: Ack - switched the language to make the documented guidelines have precedence.

File: doc/contribute/documentation/guidelines.rst
Comment (jfischer-no) [inline doc/contribute/documentation/guidelines.rst:490]: Flow text is easier to review when the columns are smaller.

**Review by jfischer-no — CHANGES_REQUESTED** [review]

File: doc/contribute/guidelines.rst
Comment (keith-zephyr) [reply doc/contribute/guidelines.rst:424]: I don't have a preference between 80 or 100 columns.  If 80 preferred/100 max is the official style, this should be explicitly stated here. The `.clang-format` file also needs updating to an 80 column limit.  Developers can disable  clang-format for longer lines on a case by case basis.

File: doc/contribute/guidelines.rst
Comment (keith-zephyr) [reply doc/contribute/guidelines.rst:435]: Speaking for myself, I use clang-format not because I'm trying to avoid learning the coding style of the project, but I use clang-format so I can focus my development effort on material design decisions.  Using automated code formatting also allows reviewers to focus on the material design.

clang-format generally does a very good job handling the various line wrap conditions and is obviously much faster than manual edits.

File: doc/contribute/guidelines.rst
Comment (nashif) [reply doc/contribute/guidelines.rst:424]: we already have `--max-line-length=100` set for checkpatch, why would we have something else for clang-format? 
This is just confusing.

**Comment by nashif** [comment]
dev-review meeting: The 80 vs 100 limit topic was discussed and there is consensus that we need just one limit and in this case and based on our current tool lconfiguration (checkpatch, clang-format) it will be 100. The 80 character limit will not be part f the guidelines.

**Review by nordicjm — DISMISSED** [review]

**Review by henrikbrixandersen — DISMISSED** [review]

**Review by stephanosio — DISMISSED** [review]

**Review by nashif — DISMISSED** [review]

**Comment by keith-zephyr** [comment]
The TSC approved the following motion by unanimous consent on 2023-02-08:
> Motion to approve having 100 characters as the line length limit for all code in the Project, and any references to 80 characters removed. Does not affect any existing code.

[TSC vote result](https://docs.google.com/spreadsheets/d/1dftY4peyiWips45wK8Sgsg-SFGdTXUNgJJUpF7faYLY/edit#gid=0&range=C3)

**Comment by stephanosio** [comment]
@jfischer-no Your change request is still pending. Please revisit.

File: doc/contribute/guidelines.rst
Comment (jfischer-no) [reply doc/contribute/guidelines.rst:435]: > clang-format generally does a very good job handling the various line wrap conditions and is obviously much faster than manual edits.

This is clearly not the case, in fact it is very bad on line wrapping and also fails badly at formatting multi-line macros. When reviewing the code, I will criticize most of the lines wrapped by clang-format. It is simply less readable, ugly, and more likely to be reformatted at every small change in the line.

Above, the mention of "rule enforcement" should be removed and rather replaced with a note that use of the clang format is entirely optional.

**Review by jfischer-no — CHANGES_REQUESTED** [review]

File: doc/contribute/guidelines.rst
Comment (keith-zephyr) [reply doc/contribute/guidelines.rst:435]: @jfischer-no I would like this change to make it in before the RC3 hard freeze.  I would appreciate if you can revisit this.  Thanks.

Comment (nashif) [reply doc/contribute/guidelines.rst:435]: @jfischer-no  please resopond to the above or remove your change for request.

File: doc/contribute/guidelines.rst
Comment (carlescufi) [reply doc/contribute/guidelines.rst:424]: I agree here with @nashif. The **maximum** line length is 100. It is OK to self-enforce lines of 80 or fewer, but the upstream rules are <=100.

File: doc/contribute/guidelines.rst
Comment (carlescufi) [inline doc/contribute/guidelines.rst:435]: @keith-zephyr can we please move those lines to a separate PR so we can unblock this one? This is pretty unrelated to the actual max line length.

Comment (keith-zephyr) [reply doc/contribute/guidelines.rst]: @jfischer-no resolved his comment, but I've modified the language to be explicit that the documented coding style takes precedence over any code formatter.s.

**Review by nashif — DISMISSED** [review]

**Review by henrikbrixandersen — DISMISSED** [review]

**Review by stephanosio — DISMISSED** [review]

**Review by MaureenHelm — DISMISSED** [review]

File: doc/contribute/guidelines.rst
Comment (jfischer-no) [inline doc/contribute/guidelines.rst:460]: Looks like this change belongs in the previous commit, also does not match the commit description.

Comment (keith-zephyr) [reply doc/contribute/guidelines.rst:460]: Good catch. Fixed.

**Review by jfischer-no — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Review by stephanosio — APPROVED** [review]

**Review by nashif — APPROVED** [review]

**Pull request merged**
